script.lua

```diff
@@ -5610,6 +5610,52 @@ function upCastPosSizRot(pos,size,rot,dist,types)
     end
     return hitObjects
 end
+-----
+local Elements = {}
+Elements.__index = Elements
+function Elements:new(init)
+    local outTable = {0,0,0,0,0,0,0,0}
+    setmetatable(outTable, self)
+    outTable:add(init)
+    return outTable
+end
+function Elements:add(other)
+    if other == nil then
+        return
+    elseif type(other) == "table" then
+        for i = 1, 8 do
+            self[i] = self[i] + other[i]
+        end
+    elseif type(other) == "string" then
+        for i = 1, string.len(other) do
+            self[i] = self[i] + math.floor(string.sub(other, i, i))
+        end
+    end
+end
+function Elements:threshold(threshold)
+    for i = 1, 8 do
+        if threshold[i] > self[i] then
+            return false
+        end
+    end
+    return true
+end
+function Elements:__tostring()
+    return table.concat(self, "")
+end
+
+local function modifyElements(params)
```

Comment:
should these 2 functions still be nested inside of other functions where they were before?

Reply:
I don't believe they ever were nested in each other before. They were *both* nested inside `setupPlayerArea()`, and being able to move `modifyElements()` out of `setupPlayerArea()` (so that the `params` variable was no longer shadowing) was the reason for moving `Elements` and its related functions outside `setupPlayerArea()`.

See https://github.com/spirit-island/spirit-island-mod/blob/a28d21461b2f6aa896f063d14869a29c232cbc86/script.lua#L5700-L5743 for how it was before I moved things outside `setupPlayerArea()`.

Reply:
derp